src/v1/controllers/TenantCreate/TenantController.ts

```diff
@@ -0,0 +1,58 @@
+import { Request, Response } from 'express';
+import logger from '../../utils/logger';
+import * as _ from 'lodash';
+import * as uuid from 'uuid';
+import tenantCreateJson from './createTenantValidationSchema.json';
+import { errorResponse, successResponse } from '../../utils/response';
+import httpStatus from 'http-status';
+import { createTenant } from '../../services/tenantService';
+import { schemaValidation } from '../../services/validationService';
+import { bulkCreateTenantBoard } from '../../services/tenantBoardService';
+
+export const apiId = 'api.tenant.create';
+
+const tenantCreate = async (req: Request, res: Response) => {
+  const requestBody = req.body;
+  const id = uuid.v4();
+  try {
+    //validating the schema
+    const isRequestValid: Record<string, any> = schemaValidation(requestBody, tenantCreateJson);
+    if (!isRequestValid.isValid) {
+      const code = 'TENANT_INVALID_INPUT';
+      logger.error({ code, apiId, requestBody, message: isRequestValid.message });
+      return res.status(httpStatus.BAD_REQUEST).json(errorResponse(id, httpStatus.BAD_REQUEST, isRequestValid.message, code));
+    }
+
+    //creating a new tenant
+    const tenantInserData = _.assign(_.omit(requestBody, ['tenant_board']), {
+      status: 'draft',
+      is_active: true,
+    });
+    const createNewTenant = await createTenant(tenantInserData);
+    logger.info({ apiId, requestBody, message: `Tenant Created Successfully with id:${_.get(createNewTenant, ['dataValues', 'id'])}` });
+
+    //create baord for tenant
+    const tenantBoard = _.get(req.body, 'tenant_board', []);
+    const tenantBoardDetails = _.map(tenantBoard, (board) => ({
+      name: board.name,
+      created_by: tenantInserData.created_by,
+      status: tenantInserData.status,
+      is_active: tenantInserData.is_active,
+      tenant_id: _.get(createNewTenant, ['dataValues', 'id']),
+    }));
+    const createBaord = await bulkCreateTenantBoard(tenantBoardDetails);
```

Comment:
how are handling errors received from these service calls?

Reply:
I've updated the service to include a `try-catch` block. If any exceptional error occurs within the service, it will throw an error with `error: true` and a respectful message. This allows the `catch` block to handle the error and display the message accordingly.